src/Common/src/Internal/Text/Utf8String.cs

```diff
@@ -112,5 +112,41 @@ public bool Equals(Utf8String other)
                 }
             }
         }
+
+        public int CompareTo(object other)
+        {
+            if (!(other is Utf8String))
+                throw new ArgumentException(nameof(other));
+
+            return Compare(this, (Utf8String)other);
+        }
+
+        private static int Compare(Utf8String strA, Utf8String strB)
+        {
+            int length = Math.Min(strA.Length, strB.Length);
+
+            unsafe
```

Comment:
Are you really sure you need unsafe code here? I'd rather we didn't add it unnecessarily.

Reply:
We are porting optimized pieces of `String` over to `Utf8String`, so there's a fair chunk of unsafe code in here to iterate strings via pointer instead of array index. We create and compare a **lot** of these things since mangled names are fundamental to the compiler. I measured the performance of Compare with unsafe pointer comparisons and another version that indexes into _value byte array and found a 25% increase in comparison speed with unsafe (measured with two 1MB strings compared 1000 times).